src/docfx/Errors.cs

```diff
@@ -99,14 +99,14 @@ public static Error InvalidTopicHref(Document relativeTo, string topicHref)
         /// <summary>
         /// In markdown-format toc, link(treated as inclusion) CAN ONLY be toc file, folder or absolute path.
         /// </summary>
-        public static Error InvalidTocHref(Document relativeTo, string tocHref, in Range range)
-            => new Error(ErrorLevel.Error, "invalid-toc-href", $"The toc href '{tocHref}' can only reference to a local TOC file, folder or absolute path", relativeTo.ToString(), range);
+        public static Error InvalidTocHref(Document relativeTo, SourceInfo<string> source)
+            => new Error(ErrorLevel.Error, "invalid-toc-href", $"The toc href '{source}' can only reference to a local TOC file, folder or absolute path", relativeTo.ToString(), source);
```

Comment:
this make me confused wht "file" and "source.file"

Reply:
@live1206 , I feel like we don't want to merge the two into once parameter. This also solves the need for `ToString` overload

Reply:
We will eventually use `File` from `SourceInfo` after we pass file into JsonUtility.Parse and fill `File` correctly into `SourceInfo`

Reply:
Added TODO

Reply:
> @live1206 , I also feel like we don't want to merge the two parameters into 1. This also solves the need for `ToString` overload

I think these two are just fit for `SourceInfo`, it should be fine to have an override `ToString`